pandas/core/arrays/string_.py

```diff
@@ -283,7 +283,8 @@ def __setitem__(self, key, value):
         super().__setitem__(key, value)
 
     def fillna(self, value=None, method=None, limit=None):
-        # TODO: validate dtype
+        if not isinstance(value, str):
```

Comment:
np.str_?

its a weird usage, but there's no reason why a user couldn't pass a n NA fill value.

Reply:
added both, including the missing value

Reply:
`np.str_` is already be covered by str (it subclasses str):

```
In [13]: arr = np.array(["a"])

In [14]: type(arr[0])
Out[14]: numpy.str_

In [15]: isinstance(arr[0], str)
Out[15]: True
```

Reply:
Ah right. Got rid of the `np.str_` check